modules/programs/info.nix

```diff
@@ -0,0 +1,76 @@
+# info.nix -- install texinfo, set INFOPATH, create `dir` file
+
+# This is a helper for the GNU info documentation system. By default,
+# the `info` command (and the Info subsystem within Emacs) gives easy
+# access to the info files stored system-wide, but not info files in
+# your ~/.nix-profile.
+
+# We set $INFOPATH to include `/run/current-system/sw/share/info` and
+# `~/.nix-profile/share/info` but it's not enough. Although info can
+# then find files when you explicitly ask for them, it doesn't show
+# them to you in the table of contents on startup. To do that requires
+# a `dir` file. NixOS keeps the system-wide `dir` file up to date, but
+# ignores home-installed packages.
+
+# So this module contains an activation script that generates the
+# `dir` for your home profile. Then when you start info (and both
+# `dir` files are in your $INFOPATH), it will *merge* the contents of
+# the two files, showing you a unified table of contents for all
+# packages. This is really nice.
+
+{ config, lib, pkgs, ... }:
+
+with lib;
+with import ../lib/dag.nix;
+
+let
+  cfg = config.programs.info;
+
+  # We want to reference $HOME in the $INFOPATH environment variable,
+  # and in the activation script. Using a bare `$HOME` is okay in the
+  # script and in environments set in the bash `.profile`, but if
+  # someone is using `pam` as the `sessionVariableSetter` the syntax
+  # seems to be `${HOME}`, according to the pam_env.conf(5) manual
+  # page. That also works in the shell, so we just use it here
+  # unconditionally. However, I HAVEN'T TESTED THIS WITH pam.
+  home = "\${HOME}";
+
+  # Encapsulate some paths and the info package; conceivably these
+  # could be options.
+  homeInfoPath = "${home}/.nix-profile/share/info";
+  sysInfoPath = "/run/current-system/sw/share/info";
+  infoPkg = pkgs.texinfoInteractive;
+  # The `pkgs.texinfo` (non-interactive) excludes the curses `info`
+  # application, but does include `install-info` which we use.
+in
+
+{
+  options = {
+    programs.info = {
+      enable = mkEnableOption "GNU Info";
+
+      homeInfoDirLocation = mkOption {
+        default = "${home}/.cache/info";
+        description =
+          "Directory in which to store the info `dir` file within your home.";
```

Comment:
The description field should be docbook code so I think you'll need the `filename` tag here:
```
description = ''
  Directory in which to store the info <filename>dir</filename> file within your home.
'';